doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,377 @@
+    - This script will be responsible for parsing and storing the most recent FS IO reads and writes from the `/proc/diskstats` file, particularly in planned reboot scenarios.
+    - These values would be stored in the `/host/pmon/storagemon/fs-stats-<<DISKNAME>>` file(s).
+
+
+**Logic for StorageCommon() get_fs_io_reads and get_fs_io_writes functions:**
+
+These two functions, `get_fs_io_reads` and `get_fs_io_writes`, are designed to retrieve the total number of disk reads and writes, respectively, by parsing the `/proc/diskstats` file. They utilize similar logic, differing only in the column index used to extract the relevant information.
+
+1. **Check for `psutil` Module**:
+   - The functions first check if the `psutil` module is available in the current environment by examining the `sys.modules` dictionary.
+
+2. **Use `psutil` Module (if available)**:
+   - If `psutil` is available:
+     - The functions retrieve disk I/O counters, specifying the disk for which to get the counters.
+     - They then get the read or write count for the specified disk using `read_count` or `write_count` respectively.
+
+3. **Fallback to Parsing Disk Stats File**:
+   - If `psutil` is not available:
+     - The functions open the `/proc/diskstats` file
+     - They read the contents of the file and iterate over each line.
+     - For each line, they check if the name of the storage disk is present.
+     - If the name of the storage disk is found in the line, they return the value at the appropriate zero-based index (3 for reads, 7 for writes).
+     - If no line contains the name of the storage disk, they save the respective values as 0.
+
+4. **Combine the previous Reads/Writes with the new values**:
+     - Then they add the new reads and writes to the fs_reads/fs_writes variables, respectively, to get the latest count
+     - These values are written to the `fs-stats-<<DISK>>` and returned to the caller
+
+
+#### **2.4.4 storagemond Class Diagram**
+
+![image.png](images/StoragemonDaemonClassDiagram.png)
+
+## **3. StateDB Schema**
+```
+; Defines information for each Storage Disk in a device
+
+key                 = STORAGE_INFO|<disk_name>  ; This key is for information about a specific storage disk - STORAGE_INFO|SDX
+
+; field             = value
+
+device_model        = STRING                    ; Describes the Vendor information of the disk                                           (Static)
+serial              = STRING                    ; Describes the Serial number of the disk                                                (Static)
+temperature_celsius = STRING                    ; Describes the operating temperature of the disk in Celsius                             (Dynamic)
+fs_io_reads         = STRING                    ; Describes the total number of filesystem reads completed successfully                  (Dynamic)
```

Comment:
Use the appropriate digit syntax e.g 1*10DIGIT  instead of STRING.

Reply:
Hi @venkatmahalingam -- curious to get your input: is there a reason this value needs to be 1*10 digit counter as opposed to string? I ask because this value is simply going to be saved to stateDB, so my understanding is that it doesn't really matter. Am I missing something here?